### PR TITLE
intial commit for gnome-text-editor on msys2

### PR DESCRIPTION
**Pull request by ZachBacon**
Just the initial package for gnome-text-editor for msys2

### CODE REVIEW

File: mingw-w64-gnome-text-editor/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+         "${MINGW_PACKAGE_PREFIX}-enchant"
+         "${MINGW_PACKAGE_PREFIX}-glib2"
+         "${MINGW_PACKAGE_PREFIX}-gtksourceview5"
+         "${MINGW_PACKAGE_PREFIX}-libadwaita")
+makedepends=("${MINGW_PACKAGE_PREFIX}-gobject-introspection"
+             "${MINGW_PACKAGE_PREFIX}-gtk-doc"
+             "${MINGW_PACKAGE_PREFIX}-gettext"
+             "${MINGW_PACKAGE_PREFIX}-itstool"
+             "${MINGW_PACKAGE_PREFIX}-meson"
+             "${MINGW_PACKAGE_PREFIX}-pkg-config"
```
Comment (Biswa96) [inline mingw-w64-gnome-text-editor/PKGBUILD:21]: Replace pkg-config with pkgconf.

File: mingw-w64-gnome-text-editor/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+             "${MINGW_PACKAGE_PREFIX}-cc")
+options=('strip' 'staticlibs')
```
Comment (Biswa96) [inline mingw-w64-gnome-text-editor/PKGBUILD:23]: These options are default and can be removed. See `grep OPTIONS /etc/makepkg_mingw.conf`

File: mingw-w64-gnome-text-editor/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+install=${_realname}-${MSYSTEM}.install
+source=(https://download.gnome.org/sources/${_realname}/${pkgver%.*}/${_realname}-${pkgver}.tar.xz)
+sha256sums=('33045c7a123faa27dee7eb9ba9a6f26f1b1718c5a0e76337d079a0d4c900e146')
+
+build() {
+  [[ -d build-${MINGW_CHOST} ]] && rm -rf build-${MINGW_CHOST}
+  mkdir -p build-${MINGW_CHOST} && cd build-${MINGW_CHOST}
+
+  MSYS2_ARG_CONV_EXCL="--prefix=" \
+  ${MINGW_PREFIX}/bin/meson.exe \
```
Comment (Biswa96) [inline mingw-w64-gnome-text-editor/PKGBUILD:35]:
```suggestion
  ${MINGW_PREFIX}/bin/meson.exe setup \
```

File: mingw-w64-gnome-text-editor/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+makedepends=("${MINGW_PACKAGE_PREFIX}-gobject-introspection"
+             "${MINGW_PACKAGE_PREFIX}-gtk-doc"
+             "${MINGW_PACKAGE_PREFIX}-gettext"
+             "${MINGW_PACKAGE_PREFIX}-itstool"
+             "${MINGW_PACKAGE_PREFIX}-meson"
+             "${MINGW_PACKAGE_PREFIX}-pkgconf"
+             "${MINGW_PACKAGE_PREFIX}-cc")
+license=('spdx:GPL-2.0-or-later')
+url="https://www.gnome.org"
+install=${_realname}-${MSYSTEM}.install
```
Comment (lazka) [inline mingw-w64-gnome-text-editor/PKGBUILD:25]: All the install files are not needed, glib2 has a hook which compiles schemas: https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-glib2/glib-compile-schemas.hook.in

**Comment by ZachBacon** [comment]
I totally was paying attention to the commit before, honest :rofl:

File: mingw-w64-gnome-text-editor/PKGBUILD (hunk 4 above)
Comment (ZachBacon) [reply mingw-w64-gnome-text-editor/PKGBUILD]: done

**Comment by Biswa96** [comment]
The dependencies are not available in clangarm64. @hmartinez82 Would you like to check if the dependencies can be built in clangarm64? Otherwise, it has be disabled here.

**Comment by ZachBacon** [comment]
crap, my dumbass just assumed it was. I'll have to see about firing up a windows 11 arm64 vm since right now my pi is in use as an sql server

**Comment by hmartinez82** [comment]
I'll look into it. What are the deps?

**Comment by Biswa96** [comment]
Each and every packages in depends or makedepends list.

**Comment by hmartinez82** [comment]
@Biswa96 I meant the ones that needs building on ARM64 :) But  I got back to my PC now, I'll look

**Comment by ZachBacon** [comment]
@hmartinez82 looks like just gtksourceview5

**Comment by hmartinez82** [comment]
Wow a rabbit hole 😅 The problem is GTK4, both gtksourceview5 and libadwaita depend on GTK4, but GTK4 depends on mingw-w64-gst-plugins-bad which depends on a bunch of codecs libs that are not build for ARM64 yet:
```
error: target not found: mingw-w64-clang-aarch64-faac
error: target not found: mingw-w64-clang-aarch64-fdk-aac
error: target not found: mingw-w64-clang-aarch64-libnice
error: target not found: mingw-w64-clang-aarch64-libsrtp
error: target not found: mingw-w64-clang-aarch64-openh264
error: target not found: mingw-w64-clang-aarch64-webrtc-audio-processing
error: target not found: mingw-w64-clang-aarch64-zbar
```

**Comment by ZachBacon** [comment]
@hmartinez82 sorry also libadwaita too

**Comment by ZachBacon** [comment]
oof... yeah

**Comment by hmartinez82** [comment]
@Biswa96 @ZachBacon Disable clangarm64 for now in this package. I'll start poking into those codecs now.

**Comment by ZachBacon** [comment]
I'll make the update now

**Pull request merged**